preview/MsixCore/MsixCoreInstaller/PrepareDevirtualizedRegistry.cpp

```diff
@@ -25,9 +25,11 @@ HRESULT PrepareDevirtualizedRegistry::ExtractRegistry(bool remove)
 {
     std::wstring registryFilePath = m_msixRequest->GetPackageDirectoryPath() + registryDatFile;
 
-    AutoPtr<RegistryDevirtualizer> registryDevirtualizer;
+    std::shared_ptr<RegistryDevirtualizer> registryDevirtualizer;
     RETURN_IF_FAILED(RegistryDevirtualizer::Create(registryFilePath, m_msixRequest, &registryDevirtualizer));
-    RETURN_IF_FAILED(registryDevirtualizer->Run(remove));
+    m_msixRequest->SetRegistryDevirtualizer(registryDevirtualizer);
+
+    //RETURN_IF_FAILED(registryDevirtualizer->Run(remove));

```

Comment:
>/RETURN_IF_FAILED(registryDevirtualizer->Run(remove)) [](start = 5, length = 53)

you can delete this.

Reply:
Moved this to ExecuteForRemoveRequest() in PrepareDevirtualizedRegistry.cpp since this should still run in the beginning for remove. Removed this line from ExtractRegistry since it should not run on add now.